extensions/wikia/ARecoveryEngine/ARecoveryModule.class.php

```diff
@@ -4,12 +4,9 @@ class ARecoveryModule {
 	/**
 	 * Checks whether PageFair recovery is enabled (on current wiki)
 	 *
-	 * $wgAdDriverEnablePageFairRecovery === false; // disabled on wiki
-	 * $wgAdDriverEnablePageFairRecovery === true; // enabled on wiki
-	 *
 	 * @return bool
 	 */
-	public function isPageFairRecoveryEnabled() {
+	public static function isPageFairRecoveryEnabled() {
 		global $wgUser, $wgAdDriverEnablePageFairRecovery;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)